### ISSUE TITLE
Configure `derive(From)` for each variant

### ISSUE THREAD

**Issue by LukasKalbertodt** [JelteF/derive_more#60]
I'd like to use `derive(From)` but the semantics is not exactly what I need. I usually want to use that for custom error types. Consider this:

```rust
enum Error {
    Io(std::io::Error),
    Parse(parse::Error),
    UnsupportedType(DataType),
    UnsupportedLength(usize),
    Other(String),
}
```

So I'd like to implement `From<std::io::Error>` and `From<parse::Error>` for this type, but I don't want to have a `From` impl for the three other types as that doesn't make a lot of sense. So it would be cool if I could annotate on the variants if they should generate an `From` impl or not. 

```rust
#[derive(From)]
enum Error {
    Io(std::io::Error),
    Parse(parse::Error),
    #[derive_more(NoFromImpl)]
    UnsupportedType(DataType),
    #[derive_more(NoFromImpl)]
    UnsupportedLength(usize),
    #[derive_more(NoFromImpl)]
    Other(String),
}
```

Or something like this.

**Comment by mmrath** [comment]
This would be super useful

**Comment by JelteF** [comment]
This shouldn't be too hard to implement anymore. I expect I will land this in the next release.

**Comment by JelteF** [comment]
This is now on master. The syntax using opt outs is like this:
```rust
#[derive(From)]
enum Error {
    Io(std::io::Error),
    Parse(parse::Error),
    #[from(ignore)]
    UnsupportedType(DataType),
    #[from(ignore)]
    UnsupportedLength(usize),
    #[from(ignore)]
    Other(String),
}
```

The syntax using opt in is like this:
```rust
#[derive(From)]
enum Error {
    #[from]
    Io(std::io::Error),
    #[from]
    Parse(parse::Error),
    UnsupportedType(DataType),
    UnsupportedLength(usize),
    Other(String),
}
```

**Comment by LukasKalbertodt** [comment]
@JelteF  Thank you :)

**Comment by JelteF** [comment]
This is now released as version 0.99.0. Announcement link: https://reddit.com/r/rust/comments/duptc0/derive_more_0990_released_biggest_release_ever
